src/firebolt/async_db/cursor.py

```diff
@@ -267,6 +272,13 @@ def _reset(self) -> None:
         self._idx = 0
         self._row_sets = []
         self._next_set_idx = 0
+        self._query_id = ""
+
+    def _query_id_from_response_async(self, response: Response) -> str:
+        if response.headers.get("content-length", "") == "0":
+            raise InternalError("No response to asynchronous query.")
+        query_data = response.json(parse_float=str)
```

Comment:
no need for `parse_float` parameter here. It's used for handling decimals in query response. We shouldn't encounter those here

Reply:
Roger that.